### PR TITLE
[DOC] Link to packed data doc

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: pack.rb [hunk 1]
```diff
@@ -1,308 +1,30 @@
 class Array
   #  call-seq:
-  #     arr.pack( aTemplateString ) -> aBinaryString
-  #     arr.pack( aTemplateString, buffer: aBufferString ) -> aBufferString
+  #    pack(template, buffer = nil) -> string
```
Comment (peterzhu2118) [inline pack.rb:3]: `buffer` is a keyword arg, not a positional one.

```ruby
irb(main):002:0> [1, 2, 3].pack("i*", "hi")
<internal:pack>:143:in `pack': wrong number of arguments (given 2, expected 1) (ArgumentError)
	from (irb):2:in `<main>'
	from /Users/peter/.rubies/ruby-3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
	from /Users/peter/.gem/ruby/3.1.2/bin/irb:25:in `load'
	from /Users/peter/.gem/ruby/3.1.2/bin/irb:25:in `<main>'
irb(main):003:0> [1, 2, 3].pack("i*", buffer: "hi")
=> "hi\u0001\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0003\u0000\u0000\u0000"
```

```suggestion
  #    pack(template, buffer: nil) -> string
```

File: pack.rb [hunk 2]
```diff
@@ -1,308 +1,30 @@
...
-  #    "abc \0abc \0".unpack('Z*Z*')       #=> ["abc ", "abc "]
-  #    "aa".unpack('b8B8')                 #=> ["10000110", "01100001"]
-  #    "aaa".unpack('h2H2c')               #=> ["16", "61", 97]
-  #    "\xfe\xff\xfe\xff".unpack('sS')     #=> [-2, 65534]
-  #    "now=20is".unpack('M*')             #=> ["now is"]
-  #    "whole".unpack('xax2aX2aX1aX2a')    #=> ["h", "e", "l", "l", "o"]
-  #
-  # This table summarizes the various formats and the Ruby classes
-  # returned by each.
+  #   unpack(template, offset = 0) -> array
```
Comment (peterzhu2118) [inline pack.rb:14]: Same here:

```suggestion
  #   unpack(template, offset: 0) -> array
```

File: pack.rb [hunk 3]
```diff
@@ -1,308 +1,30 @@
...
-  #
-  # Thus unpack1 is convenient, makes clear the intention and signals
-  # the expected return value to those reading the code.
-  #
-  # The keyword <i>offset</i> can be given to start the decoding after skipping
-  # the specified amount of bytes:
-  #   "abc".unpack1("C*") # => 97
-  #   "abc".unpack1("C*", offset: 2) # => 99
-  #   "abc".unpack1("C*", offset: 4) # => offset outside of string (ArgumentError)
+  #   unpack1(template, offset = 0) -> object
```
Comment (peterzhu2118) [inline pack.rb:24]: Ditto:

```suggestion
  #   unpack1(template, offset: 0) -> object
```

**Comment by peterzhu2118** [comment]
@BurdetteLamar Was this PR accidentally closed?

**Comment by BurdetteLamar** [comment]
Yes, accidentally.  I'm amazed to find that it can be (and has been) reopened.

    On Monday, October 17, 2022 at 04:15:24 PM CDT, Peter Zhu ***@***.***> wrote:  
 
 


@BurdetteLamar Was this PR accidentally closed?

**Review by peterzhu2118 — APPROVED** [review]

**Comment by st0012** [comment]
👋 Because `ri` can't automatically expand links, this (and similar doc refactors) will dramatically reduce the content users can get. For example, users can only see this when they do `ri Array#pack`:


```
= Array#pack

(from ruby core)
------------------------------------------------------------------------
  pack(template, buffer: nil) -> string

------------------------------------------------------------------------

Formats each element in self into a binary string; returns that string.
See {Packed Data}[rdoc-ref:packed_data.rdoc].
```

Compared to the previous:

```
= Array#pack

(from ruby core)
------------------------------------------------------------------------
  arr.pack( aTemplateString ) -> aBinaryString
  arr.pack( aTemplateString, buffer: aBufferString ) -> aBufferString

------------------------------------------------------------------------

Packs the contents of arr into a binary sequence according to the
directives in aTemplateString (see the
table below) Directives ``A,'' ``a,'' and ``Z'' may be followed by a
count, which gives the width of the resulting field. The remaining
directives also may take a count, indicating the number of array
elements to convert. If the count is an asterisk (``*''), all remaining
array elements will be converted. Any of the directives ``sSiIlL'' may
be followed by an underscore (``_'') or exclamation mark (``!'') to use
the underlying platform's native size for the specified type; otherwise,
they use a platform-independent size. Spaces are ignored in the template
string. See also String#unpack.
```

It'd be great for `ri` to support link expansion, but it's unknown how challenging it'd be. I'm also not asking you to stop similar work, but let's be aware of this potential con for similar changes.

**Pull request merged**